### PR TITLE
ORV2-2161: added initial cypress test cases

### PR DESCRIPTION
**Pull request by bcgov-brwang**
# Description

Please provide a summary of the change and the issue fixed. Please include relevant context. List dependency changes.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1327-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1327-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1327-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1327-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1327-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1327-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1327-dops.apps.silver.devops.gov.bc.ca/api)
- [Scheduler](https://onroutebc-1327-scheduler.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -71,6 +71,7 @@
     "@vitejs/plugin-react-swc": "^3.6.0",
     "@vitest/coverage-v8": "^1.3.1",
     "autoprefixer": "^10.4.17",
+    "cypress": "^13.7.2",
     "eslint": "^8.57.0",
     "eslint-config-prettier": "^9.1.0",
     "eslint-plugin-import": "^2.29.1",
```
Comment (erikataot) [inline frontend/package.json:78]: @bcgov-brwang - are we missing a `script` entry in order to launch cypress locally?

Comment (erikataot) [reply frontend/package.json:78]: maybe
`
"test:cypress": "cypress run" 
`
something like that?

Comment (erikataot) [reply frontend/package.json:78]: or `"test:e2e"` ? 🤷

File: frontend/cypress.config.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { defineConfig } from "cypress";
+
+export default defineConfig({
+  e2e: {
+    setupNodeEvents(on, config) {
+      // implement node event listeners here
+    },
+  },
+  env: {
+    CYPRESS_baseUrl: 'http://localhost:3000',
```
Comment (erikataot) [inline frontend/cypress.config.ts]: this will work on local dev...but how will it hold up moving through the environments?

**Review by cberg-aot — APPROVED** [review]
looks solid!  great first test to integrate

**Review by erikataot — APPROVED** [review]
nice

File: frontend/cypress.config.ts
Comment (cberg-aot) [inline frontend/cypress.config.ts:1]: Remove all env except username/password & url.  Refactor to import from .env

File: .github/workflows/.tests.yml
Comment (cberg-aot) [inline .github/workflows/.tests.yml:1]: @cberg-aot Remove CYPRESS_ variables.  ICEPAY_USERNAME, ICEPAY_PASSWORD & ICEPAY_URL will be added

File: frontend/cypress/e2e/home-page.cy.ts
Comment (cberg-aot) [inline frontend/cypress/e2e/home-page.cy.ts:1]: remove

File: frontend/cypress/screenshots/home-page.cy.ts/Login and start application till payment complete -- should complete an application from scratch with an existing test acount (failed).png
Comment (cberg-aot) [inline frontend/cypress/screenshots/home-page.cy.ts/Login and start application till payment complete -- should complete an application from scratch with an existing test acount (failed).png:1]: gitignore screenshots @cberg-aot

File: frontend/cypress/support/commands.ts
Comment (cberg-aot) [inline frontend/cypress/support/commands.ts:1]: Remove

File: frontend/cypress.config.ts
Comment (cberg-aot) [reply frontend/cypress.config.ts:1]: @bcgov-brwang using IDIR_ for login info

**Pull request closed without merging**